devito/dimension.py

```diff
@@ -32,6 +31,11 @@ def __new__(cls, name, **kwargs):
     def __str__(self):
         return self.name
 
+    @property
+    def dtype(self):
+        # TODO: Do dimensions really need a dtype?
```

Comment:
This should be figured out -- it feels quite an important detail. Should this rather inherit from ``devito.Symbol`` ?

Reply:
Again, largely unrelated to the point of this PR. Doesn't HAVE to be answered before this goes in.